ddp-workspace/projects/ddp-sdk/src/lib/components/prism/prism.component.ts

```diff
@@ -0,0 +1,129 @@
+  @Input() withProxy = false;
+  @Input() withInvitationId = false;
+  @Input() withLegacyId = false;
+
+  @ViewChild(MatPaginator) paginator: MatPaginator;
+  @ViewChild(MatSort) sort: MatSort;
+
+  public searchField = new FormControl();
+  public displayedColumns = BASE_COLUMNS;
+  public dataSource = new MatTableDataSource<SearchParticipant>([]);
+  public totalCount = 0;
+  public isSearchLoading = false;
+  public isSearchDebounce = false;
+  public MIN_SEARCH_LENGTH = 4;
+  private ngUnsubscribe = new Subject();
+  public readonly enrollmentStatusTypeToLabel = enrollmentStatusTypeToLabel;
+
+  constructor(
+    private sessionService: SessionMementoService,
+    private participantsSearch: ParticipantsSearchServiceAgent,
+    private router: Router) { }
+
+  public ngOnChanges(changes: SimpleChanges): void {
+    if (changes['withProxy'] || changes['withInvitationId'] || changes['withLegacyId']) {
+      this.displayedColumns = [
+          ...BASE_COLUMNS,
+          ...(changes['withInvitationId']?.currentValue ? INVITATION_COLUMNS : []),
+          ...(changes['withProxy']?.currentValue ? PROXY_COLUMNS : []),
+          ...(changes['withLegacyId']?.currentValue ? LEGACY_COLUMNS : []),
+      ];
+    }
+  }
+
+  public ngOnInit(): void {
+    this.setSubject();
+    this.initSearchListener();
+  }
+
+  public ngAfterViewInit(): void {
+    this.dataSource.paginator = this.paginator;
+    this.dataSource.sort = this.sort;
+    this.dataSource.sortingDataAccessor = (item: SearchParticipant, property: string) => {
+      switch (property) {
+        case 'userName': return item.firstName || item.lastName ? `${item.firstName } ${ item.lastName }` : '';
+        case 'proxyUserName': return item.proxy?.firstName || item.proxy?.lastName ? `${item.proxy.firstName } ${ item.proxy.lastName }` : '';
+        case 'legacyShortId': return Number(item.legacyShortId);
+        default: return item[property];
+      }
+    };
+  }
+
+  public ngOnDestroy(): void {
+    this.ngUnsubscribe.next();
+    this.ngUnsubscribe.complete();
+  }
+
+  public clearSearch(): void {
+    this.searchField.reset('', {emitEvent: true});
+  }
+
+  private setSubject(user?: SearchParticipant): void {
+    this.sessionService.setInvitationId(user?.invitationId);
+    this.sessionService.setParticipant(user?.guid);
+  }
+
+  public hasValidUserSearchQuery(): boolean {
+    const value = this.searchField.value?.trim();
+    return value && value.length >= this.MIN_SEARCH_LENGTH;
+  }
+
+  private initSearchListener(): void {
+    this.searchField.valueChanges.pipe(
+      map(query => query.trim()),
+      distinctUntilChanged(),
+      tap(() => { this.isSearchDebounce = true; }),
+      debounceTime(300),
+      tap(() => {
+        this.isSearchLoading = true;
+        this.isSearchDebounce = false;
+      }),
+      switchMap((query) => {
+        if (query.length >= this.MIN_SEARCH_LENGTH) {
+          return this.participantsSearch.search(query);
```

Comment:
What if the search returns an error ? 
I suggest (at least) adding logDebug info for errors.

Reply:
the error is handled in serviceAgent service from which ParticipantsSearchServiceAgent inherits. logError is already there